File: android/app/src/main/java/com/xirigo/ecommerce/core/designsystem/component/XGSectionHeader.kt
package com.xirigo.ecommerce.core.designsystem.component

import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.outlined.ArrowForward
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.semantics.Role
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import com.xirigo.ecommerce.R
import com.xirigo.ecommerce.core.designsystem.theme.XGColors
import com.xirigo.ecommerce.core.designsystem.theme.XGSpacing
import com.xirigo.ecommerce.core.designsystem.theme.XGTheme

/** Arrow icon size from token spec: `arrowIconSize = 12`. */
private val ArrowIconSize = 12.dp

/**
 * Section header with title, optional subtitle, and optional "See All" action.
 *
 * Token source: `shared/design-tokens/components/atoms/xg-section-header.json`
 *
 * @param title Section title text — rendered in `XGTypography.titleMedium` (18sp SemiBold).
 * @param modifier Optional [Modifier] for the root row.
 * @param subtitle Optional subtitle — rendered in `XGTypography.labelLarge` (14sp Medium).
 * @param onSeeAllClick If non-null, displays a "See All" action link with arrow icon.
 */
@Composable
fun XGSectionHeader(
    title: String,
    modifier: Modifier = Modifier,
    subtitle: String? = null,
    onSeeAllClick: (() -> Unit)? = null,
) {
    Row(
        modifier = modifier
            .fillMaxWidth()
            .padding(horizontal = XGSpacing.ScreenPaddingHorizontal),
        horizontalArrangement = Arrangement.SpaceBetween,
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Column(
            modifier = Modifier.weight(1f),
            verticalArrangement = Arrangement.spacedBy(XGSpacing.XXS),
        ) {
            Text(
                text = title,
                style = MaterialTheme.typography.titleMedium,
                color = XGColors.OnSurface,
            )
            if (subtitle != null) {
                Text(
                    text = subtitle,
                    style = MaterialTheme.typography.labelLarge,
                    color = XGColors.OnSurfaceVariant,
                )
            }
        }

        if (onSeeAllClick != null) {
            Row(
                modifier = Modifier
                    .clickable(role = Role.Button, onClick = onSeeAllClick)
                    .padding(vertical = XGSpacing.SM),
                verticalAlignment = Alignment.CenterVertically,
            ) {
                Text(
                    text = stringResource(R.string.common_see_all),
                    style = MaterialTheme.typography.labelLarge,
                    color = XGColors.BrandPrimary,
                )
                Spacer(modifier = Modifier.width(XGSpacing.XS))
                Icon(
                    imageVector = Icons.AutoMirrored.Outlined.ArrowForward,
                    contentDescription = null,
                    modifier = Modifier.size(ArrowIconSize),
                    tint = XGColors.BrandPrimary,
                )
            }
        }
    }
}

@Preview(showBackground = true)
@Composable
private fun XGSectionHeaderPreview() {
    XGTheme {
        XGSectionHeader(title = "Popular Products", onSeeAllClick = {})
    }
}

@Preview(showBackground = true)
@Composable
private fun XGSectionHeaderWithSubtitlePreview() {
    XGTheme {
        XGSectionHeader(
            title = "Categories",
            subtitle = "Browse by category",
        )
    }
}

@Preview(showBackground = true)
@Composable
private fun XGSectionHeaderNoActionPreview() {
    XGTheme {
        XGSectionHeader(title = "Daily Deal")
    }
}

File: ios/XiriGoEcommerce/Core/DesignSystem/Component/XGSectionHeader.swift
import SwiftUI

// MARK: - XGSectionHeader

/// Section title with optional subtitle and optional "See All" action link.
///
/// Token source: `shared/design-tokens/components/atoms/xg-section-header.json`
///
/// - `title`: `XGTypography.subtitle` (18pt SemiBold) + `XGColors.onSurface`
/// - `subtitle`: `XGTypography.bodyMedium` (14pt Medium) + `XGColors.onSurfaceVariant`
/// - `seeAll`: `XGTypography.bodyMedium` + `XGColors.brandPrimary`
/// - `arrowIconSize`: 12pt
/// - `horizontalPadding`: `XGSpacing.screenPaddingHorizontal`
/// - `subtitleSpacing`: `XGSpacing.xxs`
struct XGSectionHeader: View {
    // MARK: - Lifecycle

    init(
        title: String,
        subtitle: String? = nil,
        onSeeAllAction: (() -> Void)? = nil,
    ) {
        self.title = title
        self.subtitle = subtitle
        self.onSeeAllAction = onSeeAllAction
    }

    // MARK: - Internal

    var body: some View {
        HStack(alignment: .top) {
            VStack(alignment: .leading, spacing: XGSpacing.xxs) {
                Text(title)
                    .font(XGTypography.subtitle)
                    .foregroundStyle(XGColors.onSurface)

                if let subtitle {
                    Text(subtitle)
                        .font(XGTypography.bodyMedium)
                        .foregroundStyle(XGColors.onSurfaceVariant)
                }
            }

            Spacer()

            if let onSeeAllAction {
                Button(action: onSeeAllAction) {
                    HStack(spacing: XGSpacing.xs) {
                        Text(String(localized: "common_see_all"))
                            .font(XGTypography.bodyMedium)
                            .foregroundStyle(XGColors.brandPrimary)

                        Image(systemName: "chevron.right")
                            .font(.system(size: Constants.arrowIconSize, weight: .medium))
                            .foregroundStyle(XGColors.brandPrimary)
                            .accessibilityHidden(true)
                    }
                }
                .frame(minWidth: XGSpacing.minTouchTarget, minHeight: XGSpacing.minTouchTarget)
                .accessibilityLabel(
                    String(localized: "common_see_all") + " " + title,
                )
            }
        }
        .padding(.horizontal, XGSpacing.screenPaddingHorizontal)
    }

    // MARK: - Private

    private enum Constants {
        /// Token: `arrowIconSize = 12` from xg-section-header.json.
        static let arrowIconSize: CGFloat = 12
    }

    private let title: String
    private let subtitle: String?
    private let onSeeAllAction: (() -> Void)?
}

// MARK: - Previews

#Preview("XGSectionHeader") {
    VStack(spacing: XGSpacing.lg) {
        XGSectionHeader(title: "Popular Products")

        XGSectionHeader(
            title: "New Arrivals",
            subtitle: "Just added this week",
            onSeeAllAction: {},
        )

        XGSectionHeader(
            title: "Categories",
            onSeeAllAction: {},
        )
    }
    .padding(.vertical)
    .xgTheme()
}
